### PR TITLE
feat(filters): expose filter descriptions in the filter tools

### PR DESCRIPTION
**Pull request by craigcarlyle**
# Pull Request

## Short description

Filters are getting descriptions, so the filter tools can read and write them. `add-filters` sets one, `update-filters` changes or clears it, `find-filters` returns it.

- Clearing uses `"remove"` rather than null, matching how `update-tasks` clears a due date. The schema lint rejects nullable optional strings because they break the Gemini API.
- `readFilterDescription` is scaffolding. Delete it once the SDK PR lands and the field is properly typed.
- Blocked on Doist/todoist-sdk-typescript#668.

Backend: Doist/Todoist#29669

## PR Checklist

-   [x] Added tests for bugs / new features
-   [ ] Updated docs (README, etc.)
-   [ ] New tools added to `getMcpServer` AND exported in `src/index.ts`.

No new tools, so no registration or export changes.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/tools/find-filters.ts [hunk 1]
```diff
@@ -48,6 +55,7 @@ const findFilters = {
             id: f.id,
             name: f.name,
             query: f.query,
+            description: readFilterDescription(f),
```
Comment (doistbot) [inline src/tools/find-filters.ts:58]: <a href="#">P2</a>  Only update-filters.test.ts gained coverage for the new field. Per the repo's testing requirements for new tool parameters, the other two paths are untested: add-filters.test.ts has no test that a passed description is forwarded in the filter_add command and echoed in structuredContent (including the absent case at line 86), and find-filters.test.ts never exercises readFilterDescription — e.g. a mock filter with description: '' should come back with the key omitted, while a non-empty one is returned. A small test in each file would cover the sentinel's read path and the empty-string normalization.

File: src/tools/update-filters.ts [hunk 2]
```diff
@@ -15,6 +16,12 @@ const FilterUpdateSchema = z.object({
         .describe(
             'The new filter query string. Examples: "today & p1", "#Work & overdue", "@email & today".',
         ),
+    description: z
+        .string()
```
Comment (doistbot) [inline src/tools/update-filters.ts]: <a href="#">P2</a>  Accept legacy `null` here by preprocessing it to `"remove"`, as `update-tasks` does. The current string schema rejects `null` before the command-building logic runs, so existing programmatic clients that clear optional fields with `null` cannot clear a filter description.

File: src/tools/add-filters.ts [hunk 3]
```diff
@@ -14,6 +14,10 @@ const FilterSchema = z.object({
             'The filter query string. Examples: "today & p1", "#Work & overdue", "@email & today", "(p1 | p2) & !assigned". ' +
                 'Operators: | (OR), & (AND), ! (NOT), () grouping, , (multiple queries).',
         ),
+    description: z
```
Comment (doistbot) [inline src/tools/add-filters.ts]: <a href="#">P2</a>  Use the existing `optionalString` helper from `../utils/schema-helpers.js` instead of raw `z.string().optional()`. Its documented purpose is exactly this case: LLMs often send `""` for omitted optional string fields, and forwarding that empty value to the Todoist API can fail the request. `optionalString('A markdown description explaining what the filter is for.')` gives the same `string | undefined` type and normalizes `""` to `undefined`, so the existing `filter.description !== undefined` spread omits it.

File: src/tools/add-filters.ts [hunk 4]
```diff
@@ -49,6 +53,9 @@ const addFilters = {
                 {
                     name: filter.name,
                     query: filter.query,
+                    ...(filter.description !== undefined
```
Comment (doistbot) [inline src/tools/add-filters.ts:54]: <a href="#">P2</a>  The repo's testing requirements ask for setting-value coverage when adding a new tool parameter, but add-filters has no test for a description on create: nothing verifies description lands in the filter_add command args, that it's omitted from both the command and structuredContent when absent, or that an empty-string description is omitted from the output while still sent in the command. Mirror the update-filters tests: a set test asserting commands[0].args.description, and an omit test asserting the key is absent in both places.

**Review by doistbot — COMMENTED** [review]
This PR wires filter descriptions through `add-filters`, `update-filters`, and `find-filters`, using the repo's standard `"remove"` sentinel and Gemini-safe optional schemas — the scaffolding `readFilterDescription` shim is nicely documented until the SDK field is typed.

Few things worth tightening:
- Test coverage only landed for `update-filters`; `add-filters` needs set/omit tests for the new field, and `find-filters` should exercise `readFilterDescription` (empty string omitted, non-empty returned).
- `update-filters` rejects legacy `null` before the command-building logic runs — preprocess it to `"remove"` like `update-tasks` does, so existing programmatic clients can still clear the field.
- Use the existing `optionalString` helper from `schema-helpers.ts` instead of hand-rolled `z.string().optional()`; it already normalizes the empty strings LLMs send for omitted optional fields.

I also left one optional follow-up note in the details below.

<details>
<summary>Optional follow-up note (1)</summary>

- **<a href="#">P3</a> src/utils/filter-description.ts:13**: The doc comment says the API clears the description on null, but update-filters now also accepts "remove" as the clearing sentinel and translates it to null before it reaches the API. Once the SDK PR lands and this helper is deleted, a reader has no in-repo note that "remove" is the clearing sentinel for descriptions (only the input schema description carries it). Either mention "remove" in the comment or drop the now-wrong framing once the field is properly typed.

</details>

<sub>[Share Feedback](https://github.com/Doist/doistbot-app/issues/new?template=review-feedback.yml&title=Review+feedback+for+Doist%2Ftodoist-mcp&pr_url=https%3A%2F%2Fgithub.com%2FDoist%2Ftodoist-mcp%2Fpull%2F581&datadog_url=https%3A%2F%2Fapp.datadoghq.com%2Flogs%3Fquery%3Dservice%253A%2528doistbot-app-sandbox%2520OR%2520doistbot-app%2529%2520%2540meta.deliveryId%253Ab5bd69b0-a714-11f1-9f3d-0bedf15f0b9d%26agg_m%3Dcount%26agg_m_source%3Dbase%26agg_t%3Dcount%26clustering_pattern_field_path%3Dmessage%26cols%3Dhost%252Cservice%26fromUser%3Dtrue%26messageDisplay%3Dinline%26refresh_mode%3Dsliding%26storage%3Dhot%26stream_sort%3Ddesc%26viz%3Dstream%26live%3Dtrue) • [Review Logs](https://app.datadoghq.com/logs?query=service%3A%28doistbot-app-sandbox%20OR%20doistbot-app%29%20%40meta.deliveryId%3Ab5bd69b0-a714-11f1-9f3d-0bedf15f0b9d&agg_m=count&agg_m_source=base&agg_t=count&clustering_pattern_field_path=message&cols=host%2Cservice&fromUser=true&messageDisplay=inline&refresh_mode=sliding&storage=hot&stream_sort=desc&viz=stream&live=true)</sub>

File: src/tools/find-filters.ts (hunk 1 above)
Comment (craigcarlyle) [reply src/tools/find-filters.ts:58]: Addressed in https://github.com/Doist/todoist-mcp/commit/be0fd44

File: src/tools/update-filters.ts (hunk 2 above)
Comment (craigcarlyle) [reply src/tools/update-filters.ts]: Addressed in https://github.com/Doist/todoist-mcp/commit/be0fd44

File: src/tools/add-filters.ts (hunk 3 above)
Comment (craigcarlyle) [reply src/tools/add-filters.ts]: Addressed in https://github.com/Doist/todoist-mcp/commit/be0fd44

File: src/tools/add-filters.ts (hunk 4 above)
Comment (craigcarlyle) [reply src/tools/add-filters.ts:54]: Addressed in https://github.com/Doist/todoist-mcp/commit/be0fd44

File: src/utils/filter-description.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
+/**
+ * Normalizes a filter's description for structured output.
+ *
+ * A filter with no description can arrive as either `null` or `""`: the column is
```
Comment (engfragui) [inline src/utils/filter-description.ts:4]: > A filter with no description can arrive as either `null` or `""`

Why is it the case? Is this intentional that it can arrive either way? 🤔

Comment (engfragui) [reply src/utils/filter-description.ts:4]: Found some more details in the find-filters test:

> // The column is nullable and the sync view normalizes NULL to "" on the way out, so a
> // filter with no description arrives as either, and a payload predating the field has
> // no key at all. All three have to surface as undefined rather than null: structured
> // content is sanitised with removeNullFields, so a null would be stripped and then
> // fail validation against the optional output schema.

I guess I'll accept it 🤝
